File: library/src/main/java/com/pengrad/telegrambot/model/request/KeyboardButton.java
package com.pengrad.telegrambot.model.request;

import java.io.Serializable;

import com.pengrad.telegrambot.model.WebAppInfo;

/**
 * Stas Parshin
 * 06 May 2016
 */
public class KeyboardButton implements Serializable {
    private final static long serialVersionUID = 0L;

    private String text;
    private boolean request_contact;
    private boolean request_location;
    private KeyboardButtonPollType request_poll;
    private KeyboardButtonRequestUsers request_users;
    private KeyboardButtonRequestChat request_chat;
    private WebAppInfo web_app;

    public KeyboardButton(String text) {
        this.text = text;
    }

    public KeyboardButton requestLocation(boolean requestLocation) {
        request_location = requestLocation;
        return this;
    }

    public KeyboardButton requestContact(boolean requestContact) {
        request_contact = requestContact;
        return this;
    }

    public KeyboardButton requestPoll(KeyboardButtonPollType poll) {
        this.request_poll = poll;
        return this;
    }

    public KeyboardButton requestUsers(KeyboardButtonRequestUsers users) {
        this.request_users = users;
        return this;
    }

    public KeyboardButton requestChat(KeyboardButtonRequestChat chat) {
        this.request_chat = chat;
        return this;
    }

    public KeyboardButton webAppInfo(WebAppInfo webApp) {
        this.web_app = webApp;
        return this;
    }
}

File: core/src/main/kotlin/com/pengrad/telegrambot/model/request/KeyboardButton.kt
package com.pengrad.telegrambot.model.request

import java.io.Serializable

/**
 * Stas Parshin
 * 06 May 2016
 */
class KeyboardButton(private val text: String) : Serializable {
    private var request_contact: Boolean = false
    private var request_location: Boolean = false

    fun requestLocation(requestLocation: Boolean): KeyboardButton {
        request_location = requestLocation
        return this
    }

    fun requestContact(requestContact: Boolean): KeyboardButton {
        request_contact = requestContact
        return this
    }

    companion object {
        private const val serialVersionUID = 0L
    }
}
